ingest/src/main/java/com/microsoft/azure/kusto/ingest/ResourceManager.java

```diff
@@ -99,11 +110,13 @@ class RefreshIngestionResourcesTask extends TimerTask {
             public void run() {
                 try {
                     refreshIngestionResources();
-                    timer.schedule(new RefreshIngestionResourcesTask(), defaultRefreshTime);
```

Comment:
This changes the semantics, no?
What causes it to be scheduled more than once? (if there isn't an exception)?

Reply:
It shouldn't change the semantics, but after looking more deeply the existing code shouldn't cause the task to be scheduled more than once.
I do prefer the change I made, because the current approach creates a new task instance on each call (whether it succeeded or failed) instead of reusing. But GC precludes that from being a problem. So I backed this part of my change out.